### PR TITLE
Debug

### PR DESCRIPTION
**Pull request by bhrao**

### CODE REVIEW

**Review by hitharr — DISMISSED** [review]

File: src/main/java/edu/utdallas/davisbase/compiler/Compiler.java [hunk 1]
```diff
@@ -252,6 +253,12 @@ public DataType convertToDavisType(Expression type)throws CompileException{
               }
               throw new CompileException("Expected an TINYINT value, but found a BIGINT value.");
 
+            case YEAR:
+	          if (Byte.MIN_VALUE <= longValue && longValue <= Byte.MAX_VALUE) {
+	            return (byte) longValue;
```
Comment (travisclagrone) [inline src/main/java/edu/utdallas/davisbase/compiler/Compiler.java]: Why is this being returned as a `byte`?

_(I know when we serialize it it should only take up 1 byte of space, but this should be represented pre-serialization using the [`java.time.Year`](https://docs.oracle.com/javase/8/docs/api/java/time/Year.html) class.)_

File: src/main/java/edu/utdallas/davisbase/compiler/Compiler.java [hunk 2]
```diff
@@ -252,6 +253,12 @@ public DataType convertToDavisType(Expression type)throws CompileException{
               }
               throw new CompileException("Expected an TINYINT value, but found a BIGINT value.");
 
+            case YEAR:
+	          if (Byte.MIN_VALUE <= longValue && longValue <= Byte.MAX_VALUE) {
```
Comment (travisclagrone) [inline src/main/java/edu/utdallas/davisbase/compiler/Compiler.java]: Does JSqlParser parse year literals as `LongValue`?

File: src/main/java/edu/utdallas/davisbase/compiler/Compiler.java [hunk 3]
```diff
@@ -469,4 +476,24 @@ public boolean getIsNullable(String tableName, String columnName)throws CompileE
...
 
+  public void validateInsertValuesMatchesCountColumns(String tableName, int size)throws IOException, StorageException, CompileException{
+    int actualColumns = 0;
+    TableFile table = context.openTableFile(CatalogTable.DAVISBASE_COLUMNS.getName());
+    while(table.goToNextRow()){
+      if(castNonNull(table.readText(DavisBaseColumnsTableColumn.TABLE_NAME.getOrdinalPosition())).equalsIgnoreCase(tableName)){
+        actualColumns++;
+      }
+    }
+    actualColumns-=1; //subtract 1 for rowid
```
Comment (travisclagrone) [inline src/main/java/edu/utdallas/davisbase/compiler/Compiler.java]: Add the following assertion for sanity checking:

```java
assert DavisBaseColumnsTableColumn.ROWID != null : "The table 'davisbase_columns' should include a listing for the 'rowid' column for each table";
```

File: src/main/java/edu/utdallas/davisbase/host/Host.java [hunk 4]
```diff
@@ -191,19 +192,21 @@ protected void writeSelectResult(SelectResult result) throws IOException {
...
 
     if (result.getData().size() <= 0) {
       printer.println("Empty result set.");
       return;
-    }
-
-    for (SelectResultDataRow row : result.getData()) {
+    }    
+    int i = 0;
+    for (SelectResultDataRow row : result.getData()) { i = 0;
```
Comment (travisclagrone) [inline src/main/java/edu/utdallas/davisbase/host/Host.java]:
```suggestion
    for (SelectResultDataRow row : result.getData()) {
      int iColumn = 0;
```

File: src/main/java/edu/utdallas/davisbase/host/Host.java [hunk 5]
```diff
@@ -191,19 +192,21 @@ protected void writeSelectResult(SelectResult result) throws IOException {
...
+    printer.println(repeat("-", ((result.getSchema().size()) * 8) + 3)); 
 
     if (result.getData().size() <= 0) {
       printer.println("Empty result set.");
       return;
-    }
-
-    for (SelectResultDataRow row : result.getData()) {
+    }    
+    int i = 0;
```
Comment (travisclagrone) [inline src/main/java/edu/utdallas/davisbase/host/Host.java]:
```suggestion
```

File: src/main/java/edu/utdallas/davisbase/host/Host.java [hunk 6]
```diff
@@ -191,19 +192,21 @@ protected void writeSelectResult(SelectResult result) throws IOException {
...
+    for (SelectResultDataRow row : result.getData()) { i = 0;
       for (@Nullable Object value : row) {
-        printer.print(formatCellValue(10, Objects.toString(castNonNull(value))) + "|");  // NOTE `value` actually can be null, and `Objects#toString(Object)` can actually accept a null argument, but the annotated JDK is over-strict here.
+        printer.print(formatCellValue(result.getSchema().getColumnName(i).length(), Objects.toString(castNonNull(value))) + "|");  // NOTE `value` actually can be null, and `Objects#toString(Object)` can actually accept a null argument, but the annotated JDK is over-strict here.
```
Comment (travisclagrone) [inline src/main/java/edu/utdallas/davisbase/host/Host.java]:
```suggestion
        printer.print(formatCellValue(result.getSchema().getColumnName(iColumn).length(), Objects.toString(castNonNull(value))) + "|");  // NOTE `value` actually can be null, and `Objects#toString(Object)` can actually accept a null argument, but the annotated JDK is over-strict here.
```

**Review by travisclagrone — CHANGES_REQUESTED** [review]

File: src/main/java/edu/utdallas/davisbase/compiler/Compiler.java [hunk 7]
```diff
@@ -253,11 +253,12 @@ public DataType convertToDavisType(Expression type)throws CompileException{
               }
               throw new CompileException("Expected an TINYINT value, but found a BIGINT value.");
 
-            case YEAR:
-	          if (Byte.MIN_VALUE <= longValue && longValue <= Byte.MAX_VALUE) {
-	            return (byte) longValue;
+            case YEAR:         
+              long tempLongValue = longValue-2000;
+	          if (Byte.MIN_VALUE <= tempLongValue && tempLongValue <= Byte.MAX_VALUE) {
+	            return (byte) (tempLongValue);
```
Comment (travisclagrone) [inline src/main/java/edu/utdallas/davisbase/compiler/Compiler.java]: This should return an instance of `java.time.Year`, not a `byte`.

**Review by travisclagrone — CHANGES_REQUESTED** [review]

**Pull request merged**